File: Exceções - IF Quest/mago.py
from personagem import Personagem
from SemManaException import SemManaException

class Mago(Personagem):
    def __init__(self, nome, vida, forca, nivel):
        super().__init__(nome, vida, forca, nivel)
        self._mana=50

    @property
    def mana(self):
        return self._mana

    @mana.setter
    def mana(self, m):
        if m >= 0:
            self._mana = m
        else:
            print("Valor de mana inválido.")

    def ficha(self):
        print("\n====FICHA==== \nNome: " + self.nome +
              "\nVida: " + str(self.vida) +
              "\nForça: " + str(self.forca) +
              "\nNivel: " + str(self.nivel) +
              "\nMana: " + str(self.mana) +
              "\nInventario:")

        if len(self.inventario) == 0:
            print("Inventario Vazio.")
        else:
            for i in range(len(self.inventario)):
                print(self.inventario[i].descricao())
        print("===========")

    def habilidade(self):
        if self.mana < 10:
            raise SemManaException(self.mana)
        self.mana = self.mana - 10
        return "rajada arcana"

File: Exceções - IF Quest/Mago.java
class Mago extends Personagem{
    private int mana;

    public Mago(String nome, int vida, int forca, int nivel){
        super(nome, vida, forca, nivel);
        setMana(50);
    }

    public int getMana(){
        return this.mana;
    }
    public void setMana(int m){
        if (m>=0){
            this.mana = m;
        }
        else{
            throw new IllegalArgumentException("Mana inválida, informe um valor maior que 0");
        }
    }
    @Override
    public void ficha(){
        System.out.println("\n====FICHA==== \nNome: " + getNome() + "\nVida: " + getVida() + "\nForça: " + getForca() + "\nNivel: " + getNivel() + "\nMana: " + getMana() + "\nInventario:");
        if (getInventario().size() == 0){
            System.out.println("Inventario Vazio.");
        }
        else{
            for (int i=0; i<getInventario().size(); i++){
                System.out.println(getInventario().get(0).descricao());
            }
        }
        System.out.println("===========");
    }
    @Override
    public String habilidade() throws SemManaException {
    
    if (getMana() < 10) {
        throw new SemManaException(getMana());
    }
    setMana(mana-10);
    return "Rajada arcana!";
    }
}